server/models/user_model.py

```diff
@@ -41,62 +46,60 @@ def verify_hash(password, hash):
         # check given password
         return sha256.verify(password, hash)
 
-    @classmethod
-    def update_experience(cls, id, exp):
-        user = cls.query.get(id)
-        user.experience = exp
-        db.session.commit()
-
     @classmethod
     def search_experience(cls, lang_levels, reviewee_id):
         # search users the are above certain experience level, return list of
-        # users that match the requirements
+        # user_ids that match the requirements
         req_language = list(lang_levels)[0]
         req_level = int(lang_levels[req_language])
-        users = cls.query.filter(UserModel.experience != None).filter(
-            UserModel.id != reviewee_id).all()
-
-        qualified_user_ids = []
-        qualified_users = []
 
-        for user in users:
-            if (req_language in user.experience) and (int(user.experience.get(req_language)) >= req_level):
-                qualified_users.append(user)
+        qualified_experiences = db.session.query(UserModel
+                                                 ).filter(
+            UserModel.id == ExperienceModel.user_id
+        ).filter(
+            ExperienceModel.user_id != reviewee_id
+        ).filter(
+            ExperienceModel.language == req_language
+        ).filter(
+            ExperienceModel.level >= req_level
+        ).all()
 
-        # sort based on number of reviews
-        qualified_users.sort(key=lambda user: user.reviews)
+        qualified_experiences.sort(key=lambda user: user.review_count)
```

Comment:
FYI, we can actually add this to the query too (using Order by) but not necessary right now)

Reply:
added .order_by(UserModel.review_count) to end of query